`metric_calculator_test` is failing on ARM

## Description
```
=== FAIL: . TestSweep (0.00s)
make[1]: *** Waiting for unfinished jobs....
    metric_calculator_test.go:276: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/aws/metrics/metric_calculator_test.go:276
        	Error:      	"1.892346ms" is not less than or equal to "1ms"
        	Test:       	TestSweep

=== FAIL: . TestSweep (re-run 1) (0.00s)
    metric_calculator_test.go:276: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/aws/metrics/metric_calculator_test.go:276
        	Error:      	"1.9[58](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/13532271051/job/37816975925?pr=38204#step:7:59)426ms" is not less than or equal to "1ms"
        	Test:       	TestSweep
```

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/13532271051/job/37816975925?pr=38204

--- a/internal/aws/metrics/metric_calculator_test.go
+++ b/internal/aws/metrics/metric_calculator_test.go
@@ -269,11 +269,10 @@ func TestSweep(t *testing.T) {
 	}()
 
 	for i := 1; i <= 2; i++ {
-		sweepTime := <-sweepEvent
-		tickTime := time.Since(start) + mwe.ttl*time.Duration(i)
+		<-sweepEvent
+		clockTime := time.Since(start)
 		require.False(t, closed.Load())
-		assert.LessOrEqual(t, mwe.ttl, tickTime)
-		assert.LessOrEqual(t, time.Since(sweepTime), mwe.ttl)
+		assert.LessOrEqual(t, mwe.ttl*time.Duration(i), clockTime)
 	}
 	require.NoError(t, mwe.Shutdown())
 	for range sweepEvent { //nolint:revive

